src/storage-controller/src/collection_mgmt.rs

```diff
@@ -608,7 +659,21 @@ where
         consolidation::consolidate(&mut self.desired);
         consolidation::consolidate(&mut self.to_write);
 
-        self.write_to_persist(responders).await?;
+        if !self.to_write.is_empty() {
```

Comment:
I think this new condition makes it impossible to force bump the upper by appending an empty update to a differential collection. Is that fine? If it is, I think it should be documented somewhere, especially because the append-only task behaves differently.